src/data/finite/basic.lean

```diff
@@ -80,7 +75,16 @@ lemma exists_min [finite α] [nonempty α] [linear_order β] (f : α → β) :
 by { haveI := fintype.of_finite α, exact fintype.exists_min f }
 
 @[priority 100] -- see Note [lower instance priority]
-instance of_is_empty {α : Sort*} [is_empty α] : finite α := finite.of_equiv _ equiv.plift
+instance of_subsingleton {α : Sort*} [subsingleton α] : finite α :=
+begin
+  casesI is_empty_or_nonempty α,
+  { exact finite.of_equiv _ (equiv.equiv_empty α).symm },
+  { casesI (unique_iff_subsingleton_and_nonempty α).mpr ⟨‹_›, ‹_›⟩,
+    exact finite.of_equiv unit (equiv.equiv_punit α).symm }
+end
```

Comment:
The old proof still works, doesn't it?
```suggestion
finite.of_equiv _ equiv.plift
```

Reply:
No, we no longer have an instance going from `subsingleton` to `fintype`.

Reply:
There's another short proof:
```lean
finite.of_injective _ $ function.injective_of_subsingleton $ \lam x, ()
```
(writing it down from memory)